src/compass/geo_cslc.py

```diff
@@ -0,0 +1,123 @@
+
+        radar_grid = burst.as_isce3_radargrid()
+        native_doppler = burst.doppler.lut2d
+        orbit = burst.orbit
+
+        # Get azimuth polynomial coefficients for this burst
+        az_carrier_poly2d = burst.get_az_carrier_poly()
+
+        # Split the range bandwidth of the burst, if required
+        if cfg.split_spectrum_params.enabled:
+            rdr_burst_raster = range_split_spectrum(burst,
+                                                    cfg.split_spectrum_params,
+                                                    scratch_path)
+        else:
+            temp_slc_path = f'{scratch_path}/{burst_id}_{pol}_temp.vrt'
+            burst.slc_to_vrt_file(temp_slc_path)
+            rdr_burst_raster = isce3.io.Raster(temp_slc_path)
+
+        # Generate output geocoded burst raster
+        geo_burst_raster = isce3.io.Raster(
+            f'{output_path}/geo_{burst_id}_{pol}',
+            geo_grid.width, geo_grid.length,
+            rdr_burst_raster.num_bands, gdal.GDT_CFloat32,
+            cfg.geocoding_params.output_format)
+
+        # Extract burst boundaries
+        b_bounds = np.s_[burst.first_valid_line:burst.last_valid_line,
+                   burst.first_valid_sample:burst.last_valid_sample]
+
+        # Create sliced radar grid representing valid region of the burst
+        sliced_radar_grid = burst.as_isce3_radargrid()[b_bounds]
+
+        # Geocode
+        isce3.geocode.geocode_slc(geo_burst_raster, rdr_burst_raster,
+                                  dem_raster,
+                                  radar_grid, sliced_radar_grid,
+                                  geo_grid, orbit,
+                                  native_doppler,
+                                  image_grid_doppler, ellipsoid, threshold,
+                                  iters,
+                                  blocksize, dem_margin, flatten,
+                                  azimuth_carrier=az_carrier_poly2d)
+
+        # Set geo transformation
+        geotransform = [geo_grid.start_x, geo_grid.spacing_x, 0,
+                        geo_grid.start_y, 0, geo_grid.spacing_y]
+        geo_burst_raster.set_geotransform(geotransform)
+        geo_burst_raster.set_epsg(epsg)
+        del geo_burst_raster
+
+    dt = time.time() - t_start
+    info_channel.log(f'geocode burst successfully ran in {dt:.3f} seconds')
+
+
+if __name__ == "__main__":
+    '''Run geocode cslc workflow from command line'''
+    # load arguments from command line
+    geo_parser = YamlArgparse()
+
+    # Get a runconfig dict from command line argumens
+    runconfig = GeoRunConfig.load_from_yaml(geo_parser.run_config_path,
+                                            'geo_cslc_s1')
+
+    # Run geocode burst workflow
+    run(runconfig)
+
+    # Save burst metadata and runconfig parameters
+    json_path = f'{runconfig.product_path}/metadata.json'
+    with open(json_path, 'w') as f_json:
+        runconfig.to_file(f_json, 'json')
```

Comment:
It would be great to name the metadata file consistent with the geocoded product. I suggest to move these few lines to run and use f'{output_path}/geo_{burst_id}_{pol}.json' for the name of the json file.

Reply:
The polygon representing the valid geocoded burst boundary not written to metadat file by default and here is not enabled. Any reason why not writing them?

Reply:
> The polygon representing the valid geocoded burst boundary not written to metadat file by default and here is not enabled. Any reason why not writing them?

I forgot to enable polygon write; will enable it here.

Reply:
> It would be great to name the metadata file consistent with the geocoded product. I suggest to move these few lines to run and use f'{output_path}/geo_{burst_id}_{pol}.json' for the name of the json file.

@hfattahi Currently all the burst metadata is stored in a single file along with runconfig attributes. Sounds like you would like the burst metadata stored separately from runconfig attributes?